app/static/js/supervisorPortal.js

```diff
@@ -182,14 +214,83 @@ function runFormSearchQuery(button) {
 
   var inAnHour = new Date(new Date().getTime() + 60 * 60 * 1000);
   Cookies.set('lsfSearchResults', data, { expires: inAnHour })
+  if (view === 'advanced') {
+    createDataTable(data)
+  } else {
+    fetchSimpleView(data)
+  }
+}
+
+function switchViewButton(view) {
+  if (view == 'advanced') {
+    $('#switchViewButton').val('simple')
+    $('#switchViewButton').html('Switch To Advanced View')
+    $('#columnPicker').empty();
+    let columns = Object.keys(simpleColumnFieldMap)
+    columns.forEach((column) => {
+      var option = $('<option>', {
+        value: simpleColumnFieldMap[column][0][1],
+        text: column
+      });
+      $('#columnPicker').append(option)
+    })
+  } else {
+    $('#switchViewButton').val('advanced')
+    $('#switchViewButton').html('Switch To Simple View')
+    $('#columnPicker').empty();
+    let columns = Object.keys(advancedColumnFieldMap)
+    columns.forEach((column) => {
+      var option = $('<option>', {
+        value: advancedColumnFieldMap[column][0][1],
+        text: column
+      });
+      $('#columnPicker').append(option)
+    })
```

Comment:
Let's pull out 240-248 into a resetColumnPicker function (or some other name), as it is basically identical to the code in the other branch (228-236). That will make each side of the conditional look a lot simpler, and we won't repeat the code.